controller/setting_controller.go

```diff
@@ -282,9 +282,9 @@ func (sc *SettingController) updateBackupstorePollInterval() (err error) {
 	if sc.bsMonitor.pollInterval == time.Duration(interval)*time.Second {
 		return nil
 	}
-
-	sc.bsMonitor.Stop()
-
+	if sc.bsMonitor.pollInterval != time.Duration(0) {
```

Comment:
Do it in the `bsMonitor.Stop()` instead. The caller shouldn't care about when the start or stop should be executed. The caller only intent to start or stop it, so if it's not needed or cannot be done, it's e.g. `monitor` who should decide what to do .